software/known-issues.md

```diff
@@ -123,3 +131,107 @@ KNOWN_CURSORS=["HORIZON"]
 5. call "newdb" on the core instance to reset core's database
 6. start up stellar-core normally, wait for it to be in "Synced!" state
 7. start up Horizon, it should be able to see that core has data to ingest
+
+# Backfilling history
+
+You have a core instance that works properly (close ledgers), but you realize
+that for some reason its history is incomplete, in the sense that the oldest
+ledger available for Horizon to import is not old enough.
+
+Reasons that can lead to this include some of the problems that can lead to
+[Gaps Detected](#gaps-detected).
+
+## Check your configuration
+
+Follow the steps in [mismatched core/Horizon configurations](#mismatched-corehorizon-configurations)
+
+In particular, if the reason you have missing data is because of
+garbage collection, you want to make sure that it doesn't happen again in the
+future!
+
+## Recovery process
+
+Important, before attempting any recovery process:
+* make sure that your configuration is correct (and if not adjust it)
+* **BACKUP** all data on your core node
+
+### Reset everything
+
+The easiest way to recover is to [reset everything](#reset-everything).
+
+This will take some time (for machines) but doesn't require manual intervention.
+
+### Manually backfill history
+
+The general idea behind this method is to construct the missing data on a
+separate stellar-core instance, followed by an "import" into the real instance.
+
+Note that **Horizon does not support backfilling**: it will have to reingest all
+data from core.
+
+#### Preparation steps
+
+1. Stop Horizon.
+2. Set the `HORIZON` cursor to 0 (as to stop garbage collection from doing anything)
+```
+stellar-core --conf X.conf 'setcursor?id=HORIZON&cursor=0'
+```
+3. Stop your stellar-core instance
+4. Connect to your core database, and run
+```sql
+SELECT ledgerseq FROM ledgerheaders ORDER BY ledgerseq ASC LIMIT 1;
+```
+this will return the _smallest_ ledger that your core instance knows about.
+
+We'll call this value `ORIGINAL_LOW_LEDGER` for now on.
+
+We'll call `LOW_LEDGER` the value that you want in history.
+
+#### Reconstruction steps
+
+Steps here are going to reconstruct history for the range `LOW_LEDGER .. ORIGINAL_LOW_LEDGER`
+
+1. Prepare a fresh new core instance with a similar configuration file
+    * use a different database, sqlite even
+    * don't start it after running `newdb`
+2. Perform a command line catchup to replay a range of ledger that includes the target range
+```
+stellar-core --conf X.conf --catchup-at LOW_LEDGER --catchup-to ORIGINAL_LOW_LEDGER
+```
+
+If this succeeds you can proceed to the next step
+
+#### Merging steps
+
+Repeat the following steps for the following sql tables:
+* `ledgerheaders` (`ledgerseq`)
+* `txhistory` (`ledgerseq`)
+* `txfeehistory` (`ledgerseq`)
+not imported from history as of this writing:
```

Comment:
```suggestion
Not imported from history as of this writing:
```

Reply:
Does this also mean that you don't need to do this process for these tables? If so, we should make that explicit.